test/server/server_test.cc

```diff
@@ -46,27 +47,29 @@ class TestComponentFactory : public ComponentFactory {
 // Class creates minimally viable server instance for testing.
 class ServerInstanceImplTest : public testing::Test {
 protected:
-  ServerInstanceImplTest()
-      : options_(std::string("test/config/integration/server.json")),
-        server_(options_, hooks_, restart_, stats_store_, fakelock_, component_factory_,
-                local_info_) {}
+  ServerInstanceImplTest() {
+    options_.reset(new testing::NiceMock<MockOptions>(TestEnvironment::temporaryFileSubstitutePorts(
```

Comment:
Yeah, no need to change to pointers, I had first at thought the port map construction would be more involved, but it's not.